global_helpers/panther.py

```diff
@@ -121,35 +127,25 @@ def increment_counter(key: str, val: int = 1) -> int:
     Returns:
         The new value of the count
     """
-    table = kv_table()
-    try:
-        response = table.update_item(
-            Key={'key': key},
-            ReturnValues='UPDATED_NEW',
-            # You can only increment attributes which already exist
-            ConditionExpression=Attr(_COUNT_COL).exists(),
-            UpdateExpression='SET #col = intCount + :incr',
-            ExpressionAttributeNames={'#col': _COUNT_COL},
-            ExpressionAttributeValues={':incr': val})
-
-        # Numeric values are returned as decimal.Decimal
-        return response['Attributes'][_COUNT_COL].to_integral_value()
-    except botocore.exceptions.ClientError as ex:
-        if ex.response['Error']['Code'] != 'ConditionalCheckFailedException':
-            raise
-
-    # The conditional check failed, meaning this item doesn't exist yet. Add it!
-    table.put_item(Item={'key': key, _COUNT_COL: val})
-    return val
+    response = kv_table().update_item(
+        Key={'key': key},
+        ReturnValues='UPDATED_NEW',
+        UpdateExpression='ADD #col :incr',
+        ExpressionAttributeNames={'#col': _COUNT_COL},
+        ExpressionAttributeValues={':incr': val},
+    )
+
+    # Numeric values are returned as decimal.Decimal
+    return response['Attributes'][_COUNT_COL].to_integral_value()
 
 
 def reset_counter(key: str) -> None:
     """Reset a counter to 0."""
     kv_table().put_item(Item={'key': key, _COUNT_COL: 0})
 
 
-def set_counter_expiration(key: str, epoch_seconds: int) -> None:
-    """Configure the counter to automatically expire at the given time.
+def set_key_expiration(key: str, epoch_seconds: int) -> None:
```

Comment:
Works for both string sets and counters (which can coexist under the same key), renaming for generality